src/main/java/carpet/mixins/RedstoneWireBlockMixin.java

```diff
@@ -64,96 +55,73 @@ private void onRedstoneWireBlockCTOR(Block.Settings settings, CallbackInfo ci) {
 
     // =
 
-    public BlockState fastUpdate(World world, BlockPos pos, BlockState state, BlockPos source) {
+    public void fastUpdate(World world, BlockPos pos, BlockState state, BlockPos source) {
         // [CM] fastRedstoneDust -- update based on carpet rule
         if (CarpetSettings.fastRedstoneDust) {
-            return wireTurbo.updateSurroundingRedstone(world, pos, state, source);
+            wireTurbo.updateSurroundingRedstone(world, pos, state, source);
+            return;
         }
-        return update(world, pos, state);
+        update(world, pos, state);
     }
 
     /**
      * @author theosib, soykaf, gnembon
-     *
-    @Inject(method = "updateLogic", at = @At("HEAD"), cancellable = true)
-    private void updateLogicAlternative(World world, BlockPos pos, BlockState state, CallbackInfoReturnable<BlockState> cir)
-    {
-        if (CarpetSettings.fastRedstoneDust)
-            cir.setReturnValue(updateLogicPublic(world, pos, state));
+     */
+    @Inject(method = "update", at = @At("HEAD"), cancellable = true)
+    private void updateLogicAlternative(World world, BlockPos pos, BlockState state, CallbackInfo cir) {
+        if (CarpetSettings.fastRedstoneDust) {
+            updateLogicPublic(world, pos, state);
+            cir.cancel();
+        }
     }
 
     @Override
     public BlockState updateLogicPublic(World world_1, BlockPos blockPos_1, BlockState blockState_1) {
```

Comment:
This method looks pretty much identical to vanilla right now, is that true? Then what's the reason for mixin into 'update'